r-package/survlearners/R/dgps.R

```diff
@@ -1,851 +1,44 @@
-#'   \item  "type1": T is drawn from an accelerated failure time model and C from a Cox model (scenario 1 in https://arxiv.org/abs/2001.09887)
-#'   \item  "type2": T is drawn from a proportional hazard model and C from a accelerated failure time (scenario 2 in https://arxiv.org/abs/2001.09887)
-#'   \item  "type3": T and C are drawn from a Poisson distribution  (scenario 3 in https://arxiv.org/abs/2001.09887)
-#'   \item  "type4": T and C are drawn from a Poisson distribution  (scenario 4 in https://arxiv.org/abs/2001.09887)
-#'   \item  "type5": is similar to "type2" but with censoring generated from an accelerated failure time model.
-#' }
 #' @param n The number of samples.
-#' @param p The number of covariates.
+#' @param p The total number of covariates.
+#' @param p_b The number of covariates in the main effect function.
+#' @param p_i The number of covariates in the interaction term.
+#' @param f_b Function form of the main effects (linear/nonlinear).
+#' @param f_i Function form of the interaction term (linear/nonlinear).
+#' @param times The time of interest.
+#' @param pi The probability of receiving treatment.
+#' @param beta The coefficient of main effect.
+#' @param gamma The coefficient of treatment-covariate interaction.
+#' @param rho The shape parameter of a Weibull distribution for censoring time.
+#' @param cen_scale The scale parameter of a Weibull distribution for censoring time.
+#' @param cenM The complexity of censoring mechanism (independent/dependent of covariates).
 #' @param Y.max The maximum follow-up time (optional).
-#' @param X The covariates (optional).
 #' @param n.mc The number of monte carlo draws to estimate the treatment effect with. Default is 10000.
 #' @param dgp The type of DGP.
 #'
 #' @return A list with entries:
 #'  `X`: the covariates, `Y`: the event times, `W`: the treatment indicator, `D`: the censoring indicator,
-#'  `cate`: the treatment effect estimated by monte carlo, `cate.sign`: the true sign of the cate for ITR comparison,
+#'  `catesp`: the treatment effect estimated by monte carlo, `catesp.sign`: the true sign of the cate for ITR comparison,
 #'  `dgp`: the dgp name, `Y.max`: the maximum follow-up time.
 #'
 #' @examples
 #' \donttest{
 #' # Generate data
 #' n <- 1000
 #' p <- 5
-#' data <- generate_causal_survival_data(n, p)
-#' # Get true CATE on a test set
-#' X.test <- matrix(seq(0, 1, length.out = 5), 5, p)
-#' cate.test <- generate_causal_survival_data(n, p, X = X.test)$cate
+#' data <- generate_causal_survival_data(n, p, p_b = 1, p_i = 1, f_b = "L", f_i = "L", times = 0.2)
+#' # Get true CATE in terms of difference in survival probabilities at time 0.2
+#' cate <- generate_causal_survival_data(n, p, p_b = 1, p_i = 1, f_b = "L", f_i = "L", times = 0.2)$cate
 #' }
 #'
 #' @export
-
-# Simulation from Recursively Imputed Survival Trees (Zhu and Kosorok 2012)
-library(mvtnorm)
-generate_survival_tree_data <- function(n, p, Y.max = NULL, X = NULL, n.mc = 10000, times = NULL,
```

Comment:
why are you deleting this here? the purpose of this PR is to "Check example for each function by changing to \donttest"

Reply:
because I did not add an example for each dgp function... but I was worried it won't pass the test if I don't add an example. I was going to delete them in a separate PR, but just got a bit lazy.

Reply:
none of the `@examples` use generate_survival_tree_data

Reply:
yes, I stopped using generate_survival_tree_data for a long while, do not want to include it in the package

Reply:
Yes, so why are you deleting it here

Reply:
I thought for all the code to pass the check, I need to add an example for every function; however, I don't want to add an example for something I am not even going to use later... instead of deleting it in a separate PR, I deleted it here

Reply:
Each estimator's  `@example` should have a dead-simple example X;Y;W;D = ...  like 

```
n <- 500
p <- 5
X <- matrix(runif(n * p), n, p)
W <- rbinom(n, 1, 0.5)
failure.time <- pmin(rexp(n) * X[, 1] + W, 1)
censor.time <- 2 * runif(n)
Y <- round(pmin(failure.time, censor.time), 2)
D <- as.integer(failure.time <= censor.time)

# run the estimator on this and predict
```
I don't see what  dgps.R has to do with that

So far in this PR that has revealed many typos/errors you needed to fix in the estimator R code, if you want to seem them you have to sift through  more than **one thousand**  diff lines:

![Screen Shot 2022-04-09 at 08 44 32](https://user-images.githubusercontent.com/7185264/162581342-6d49417b-7e62-4579-a02c-f7085164c45d.png)

most of which have nothing to do with the bugs you are uncovering in this PR 

Can that make you appreciate  doing only specific things in every PR?

Reply:
if you mean you don't want to add `@examples` to the files in dgps.R since they will be deleted you can skip that (or delete them in a separate PR), I only care about the estimators `@ examples`

Reply:
ok, sorry about that. Would you like me to close this PR and start all over again?

Reply:
You decide if you want to start a new PR or try to clean up this one. PS: doing "good separated" commits even in PRs can be useful, for example:

if you had a single commit that did:

-"Update all dgps.R" 

you could have easily undone that by doing 

git revert <the commit ID of <Update all dgps.R">

Reply:
thank you. I will work on it.

Reply:
Here is one suggestion that will make this clean and less likely to cause future bugs/mix up things:

1) Make one PR: "Add working \donntest `@examples` to all estimators"

that will not run due to various errors a), b) c) d) etc
identify those errors and fix them in a new pr:

2) Fix error a)
     Descr: Fix naming of dataW instead of data$W

3) Fix error b)
     Descr: fix something else

etc you decide how you want to split up a) , b) c) etc. (Fixing many bugs in a single commit can sometimes be dangerous because sometimes they "cancel out", maybe the the fix causes a new bug)

And merge those changes into 1) by clicking "Update branch", after everything is fixed 1) will pass

Then the MASTER [commit history](https://github.com/som-shahlab/survmetabench/commits/master) will look like:


-"Add working \donntest `@examples` to all estimators"
-"Fix error a"
-"Fix error b"
-etc etc

and you can look at that history and immediately without clicking any details understand what changes you've made to the estimators.